kerchunk/df.py

```diff
@@ -25,7 +28,182 @@ def _proc_raw(r):
+            )
+            if chunk_sizes.size == 0:
+                chunk_sizes = np.array([0])
+            nchunks = int(np.product(chunk_sizes))
+            extra_rows = row_group_size - nchunks % row_group_size
+            output_size = nchunks + extra_rows
+            # Initialize dataframe columns. Don't know why but int64 gives better
+            # compression ratio than int32
+            paths = np.full(output_size, np.nan, dtype="O")
+            offsets = np.zeros(output_size, dtype="int64")
+            sizes = np.zeros(output_size, dtype="int64")
+            raws = np.full(output_size, np.nan, dtype="O")
+            nmissing = 0
+            for metakey in [".zarray", ".zattrs"]:
+                key = f"{field}/{metakey}"
+                _write_json(os.path.join(field_path, metakey), refs[key])
+            for i, ind in enumerate(np.ndindex(tuple(chunk_sizes.astype(int)))):
+                chunk_id = ".".join([str(ix) for ix in ind])
+                key = f"{field}/{chunk_id}"
+                # Make note if expected number of chunks differs from actual
+                # number found in references
+                if key in refs:
+                    data = refs[key]
+                    if isinstance(data, list):
+                        paths[i] = data[0]
+                        offsets[i] = data[1]
+                        sizes[i] = data[2]
+                    else:
+                        raws[i] = _proc_raw(data)
+                else:
+                    nmissing += 1
+
+            if nmissing:
+                print(
+                    f"Warning: Chunks missing for field {field}. "
+                    f"Expected: {nchunks}, Found: {nchunks - nmissing}"
+                )
+            # The convention for parquet files is
+            # <url>/<field_name>/refs.parq
+            out_path = os.path.join(field_path, "refs.parq")
+            df = pd.DataFrame(
+                dict(path=paths, offset=offsets, size=sizes, raw=raws), copy=False
+            )
+            # Engine specific kwarg conventions. Set stats to false since
+            # those are currently unneeded.
+            if engine == "pyarrow":
+                # TODO: perhaps set some other defaults to pyarrow engine for
+                # optimization?
+                kwargs.update(write_statistics=False, row_group_size=row_group_size)
+            else:
+                kwargs.update(
+                    row_group_offsets=row_group_size,
+                    object_encoding=dict(raw="bytes", path="utf8"),
+                    has_nulls=["path", "raw"],
+                    stats=False,
+                )
+            df.to_parquet(
+                out_path,
+                engine=engine,
+                storage_options=storage_options,
+                compression=compression,
+                index=False,
+                **kwargs,
+            )
+
+
+def refs_to_dataframe_full(
```

Comment:
Wondering how the two functions should be named. I think it would be great to keep this version around for combining multiple reference sets since it should be quicker and simpler to use for that purpose.